### PR TITLE
Gsm driver rework, memory optimization, code clean up

### PR DESCRIPTION
**Pull request by rspolyak**

### CODE REVIEW

**Comment by rspolyak** [comment]
For review

File: bsp/gtrack/bsp.c [hunk 1]
```diff
@@ -57,24 +57,40 @@ void bspSystemPowerOff(void)
   return;
 }
 
+static void pwrOff(void *arg) {
```
Comment (garyachy) [inline bsp/gtrack/bsp.c]: Please use a naming convention

File: bsp/gtrack/bsp.c [hunk 2]
```diff
@@ -57,24 +57,40 @@ void bspSystemPowerOff(void)
...
+
+      /* release PWRKEY (automatically raises HIGH) */
+      palClearPad(GPIOC, BSP_GSM_PWR_PIN);
+
+      /* switch off device */
+      bspSystemPowerOff();
+  }
+}
+
 static void bspExtcb1(EXTDriver *extp, expchannel_t channel)
```
Comment (garyachy) [inline bsp/gtrack/bsp.c]: Could you investigate a possibillity to reuse ext_api.c from drivers folder?

File: bsp/gtrack/bsp.c [hunk 3]
```diff
@@ -158,9 +201,27 @@ RV_t bspInitComplete(void)
...
-    gptStartContinuous(&GPTD4, 3000); /* 3000 / 1000 = trigger rate in seconds */
+     * Blink status LED each 10 sec */
+    bspStartTimer3Sec();
 
     return RV_SUCCESS;
 }
+
+RV_t bspIndicateError(uint32_t blinkTime)
+{
+  gptStopTimer(&GPTD4);
```
Comment (garyachy) [inline bsp/gtrack/bsp.c:172]: Please consider creating a generic driver for timer inside driver folder

File: drivers/include/gsm_common.h [hunk 4]
```diff
@@ -34,7 +34,7 @@ typedef enum
   FINISH_STATE
 } gsm_parse_state;
 
-typedef RV_t (*gsmCmdAnalyze_t)(char *buf, uint32_t len, uint32_t *val);
+//typedef RV_t (*gsmCmdAnalyze_t)(char *buf, uint32_t len, uint32_t *val);
```
Comment (garyachy) [inline drivers/include/gsm_common.h]: Dead code?

**Review by garyachy — APPROVED** [review]

File: drivers/include/gsm_common.h (hunk 4 above)
Comment (rspolyak) [reply drivers/include/gsm_common.h]: will remove

File: bsp/gtrack/bsp.c (hunk 3 above)
Comment (rspolyak) [reply bsp/gtrack/bsp.c:172]: will log a task and handle later

File: bsp/gtrack/bsp.c (hunk 2 above)
Comment (rspolyak) [reply bsp/gtrack/bsp.c]: will do

**Pull request merged**